### PR TITLE
content-disposition header case sensitivity

### PR DESCRIPTION
**Pull request by MrSagarShah**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dartlang.org/packages/dio)
- [x] I have searched for a similar pull request in the [project](https://github.com/flutterchina/dio/pulls) and found none
- [x] I have updated this branch with the latest `develop` to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I am adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests and they pass

This merge request fixes / refers to the following issues: ...

### Pull Request Description

...

### CODE REVIEW

**Comment by MrSagarShah** [comment]
Ideally yes, I would suggest keeping it flexible.

**Comment by MrSagarShah** [comment]
Update the PR, 
Kept it as optional argument
Please have a look and let me know if anything needs to be updated.
Thanks

File: dio/lib/src/form_data.dart [hunk 1]
```diff
@@ -30,20 +30,23 @@ class FormData {
   /// Whether [finalize] has been called.
   bool get isFinalized => _isFinalized;
   bool _isFinalized = false;
+  final bool _camelCaseContentDisposition;
 
-  FormData() {
+  FormData({bool camelCaseContentDisposition = false}) {
+    _camelCaseContentDisposition = camelCaseContentDisposition;
```
Comment (AlexV525) [inline dio/lib/src/form_data.dart:36]: This is invalid since `_camelCaseContentDisposition` is `final`.

File: dio/lib/src/form_data.dart [hunk 2]
```diff
@@ -30,20 +30,23 @@ class FormData {
...
     _init();
   }
 
   /// Create FormData instance with a Map.
-  FormData.fromMap(
-    Map<String, dynamic> map, [
+  FormData.fromMap(Map<String, dynamic> map, [
     ListFormat collectionFormat = ListFormat.multi,
+  {bool camelCaseContentDisposition = false}
```
Comment (AlexV525) [inline dio/lib/src/form_data.dart:43]: This is impossible when positioned optional arguments are set.

**Review by AlexV525 — CHANGES_REQUESTED** [review]
I don't think your recent changes to the code can pass static analysis. Though I'll try to merge this into the fork without extra steps from your side.

File: dio/lib/src/form_data.dart (hunk 2 above)
Comment (MrSagarShah) [reply dio/lib/src/form_data.dart]: you are right, my mistake

File: dio/lib/src/form_data.dart (hunk 1 above)
Comment (MrSagarShah) [reply dio/lib/src/form_data.dart]: updated

**Comment by AlexV525** [comment]
Hi @MrSagarShah. We've made our hardfork repo public and published a new version of `dio`, named `diox`.
The new package contains the PR of the fix.
Please refer to https://pub.dev/packages/diox/versions/5.0.0-dev.1 to use the fork.
You can also see why we're working for a hardfork at https://github.com/cfug/diox/issues/29 and https://github.com/flutterchina/dio/issues/1607.

**Comment by MrSagarShah** [comment]
@AlexV525 Thanks, Mate. 
https://github.com/cfug/diox have more secure ways. 
Will try it out.

**Pull request closed without merging**